_posts/2018-02-16-running-foreman-with-postgresql-96-from-software-collections.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: Running Foreman with PostgreSQL 9.6 from Software Collections
+date: 2018-02-16 11:25:10
+author: Evgeni Golov
+tags:
+- foreman
+- postgresql
+---
+
+When deploying Foreman on CentOS 7, you by default get PostgreSQL 9.2 as packaged in CentOS (and RHEL).
+However PostgreSQL 9.2 is [officially EOL upstream](https://www.postgresql.org/support/versioning/) and newer versions bring various performance improvements which might benefit the Foreman installation.
+This post will show how to deploy Foreman with PostgreSQL 9.6 from [SoftwareCollections.org](https://www.softwarecollections.org).
+
+<!--more-->
+
+Given the Foreman Installer will setup everything for us, we really only have to tell the Installer that we want a different version of PostgreSQL, where to find it and how it should be configured.
+
+As the Installer does not have any command line options to configure PostgreSQL, we will have to create a `custom-hiera.yaml` with the required values.
+
+Under the hood, the Installer is using the [puppetlabs-postgresql](https://github.com/puppetlabs/puppetlabs-postgresql/) Puppet module, so that is the source for our customization.
+Initially I wanted to override the `postgresql::server::*` and `postgresql::client::*` values, as documented in the `README.md`, however doing so would require overriding *many* values, as the module generates many variables in `postgresql::params` based on the values from `postgresql::globals`.
+Instead we will override the values directly in `postgresql::globals` and let all the other classes inherit our definitions.
+
+Given that packages from [SoftwareCollections.org](https://www.softwarecollections.org) always come with a prefix for almost everything, we'll have to override the package names, systemd unit names and the data folder.
+
+Luckily, the [PostgreSQL 9.6](https://www.softwarecollections.org/en/scls/rhscl/rh-postgresql96/) software collection also contains `syspaths` packages (`rh-postgresql96-postgresql-syspaths`, `rh-postgresql96-postgresql-server-syspaths`, etc), so that we won't have to worry about scripts expecting `/usr/bin/psql` or similar to exist ("normal" SCL packages don't touch `/usr/bin` and require a `scl enable NAME` to add the SCL-specific `bin` folder to your `PATH`).
+
+Looking at [`postgresql::globals`](https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/globals.pp) and [`postgresql::params`](https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/params.pp), we come up with the following settings:
+
+```yaml
+postgresql::params::version: 9.6
```

Comment:
This can also be set via ```globals```

Reply:
Yes and no, `globals` has the "default" version, here I am explicitly saying, I am not installing the default.

Reply:
Oooh, there are two. /me blind.